content/en/database_monitoring/setup_sql_server/troubleshooting.md

```diff
@@ -127,44 +141,91 @@ If you are using a driver **other than `MSOLEDBSQL` 2019**, this error can be re
       driver: '{ODBC Driver 17 for SQL Server}'
   ```
 
-### SQL Server unable to connect 'SSL Security error (18)'
+### SQL Server unable to connect 'SSL Security error (18)' {#ssl-security-error}
 
 This is a known issue for older versions of the SQL Server ODBC driver. You can check which version of the driver is being used by the agent by looking at the connection string in the error message.
 
 For example, if you see `Provider=SQL Server` in the connection string of the error message, upgrading to a newer version of the ODBC driver will fix the error.
 
 This issue is described in more detail in this [Microsoft blog post][9]
 
+### Empty connection string {#empty-connection-string}
 
-### SQL Server 'Unable to connect: Adaptive Server is unavailable or does not exist'
+Datadog's SQL Server check relies on the adodbapi Python library, which has some limitations in the characters that it is able to use in making a connection string to a SQL Server. If your Agent experiences trouble connecting to your SQL Server, and if you find errors similar to the following in your Agent's collector.logs, your `sqlserver.yaml` probably includes some character that causes issues with adodbapi.
 
-This error can sometimes just be the result of not properly setting the `host` field. For the integration, you should the `host` field with the following syntax: `host:server,port`.
+```text
+OperationalError: (KeyError('Python string format error in connection string->',), 'Error opening connection to ""')
+```
 
-For example:
+At the moment, the only character known to cause this specific connectivity issue is the `%` character. If you want to use the "%" character in your `sqlserver.yaml`, that is if your Datadog SQL Server user password includes a `%`), you need to escape that character by including a double `%%` in place of each single `%`.
 
-```
-host: sqlserver-foo.cfxxae8cilce.us-east-1.rds.amazonaws.com
-```
-Should be set as:
-```
-host: sqlserver-foo.cfxxae8cilce.us-east-1.rds.amazonaws.com,1433
+## Diagnosing common SQL Server driver issues {#common-driver-issues}
+
+### Data source name not found, and no default driver specified {#data-source-name-not-found}
+
+This is a common error seen on linux when using the default setting for the ODBC driver. This can happen due the [DSN][10], which is set for your driver in the `/etc/odbcinst.ini` file, not matching the name of the driver that is set in your agent config.
```

Comment:
```suggestion
This is a common error seen on Linux when using the default setting for the ODBC driver. This can happen due the [DSN][10], which is set for your driver in the `/etc/odbcinst.ini` file, not matching the name of the driver that is set in your agent config.
```